.github/workflows/merge.yml

```diff
@@ -723,6 +723,120 @@ jobs:
+        with:
+          action: start
+          interval: 2000
+          node-path: target/release/safenode
+          faucet-path: target/release/faucet
+          platform: ubuntu-latest
+          build: true
+
+      - name: Check we're _not_ warned about using default genesis
+        run: |
+          if rg "USING DEFAULT" "${{ matrix.safe_path }}"/*/*/logs; then
+            exit 1
+          fi
+        shell: bash
+
+      - name: Move built binaries and clear out target dir
+        shell: bash
+        run: |
+          mv target/release/faucet ~/faucet
+          mv target/release/safe ~/safe
+          rm -rf target
+
+      - name: Check SAFE_PEERS was set
+        shell: bash
+        run: |
+          if [[ -z "$SAFE_PEERS" ]]; then
+            echo "The SAFE_PEERS variable has not been set"
+            exit 1
+          else
+            echo "SAFE_PEERS has been set to $SAFE_PEERS"
+          fi
+
+      - name: Create and fund a wallet first time
+        run: |
+          ~/faucet --log-output-dest=data-dir send 100000000 $(~/safe --log-output-dest=data-dir wallet address | tail -n 1) | tail -n 1 1>first.txt
+          echo "----------"
+          cat first.txt
+        env:
+          SN_LOG: "all"
+        timeout-minutes: 5
+
+      - name: Create and fund a wallet second time
+        run: |
+          ls -l /home/runner/.local/share
+          ls -l /home/runner/.local/share/safe
+          rm -rf /home/runner/.local/share/safe/test_faucet
+          rm -rf /home/runner/.local/share/safe/test_genesis
+          rm -rf /home/runner/.local/share/safe/client
+          ~/faucet --log-output-dest=data-dir send 100000000 $(~/safe --log-output-dest=data-dir wallet address | tail -n 1) | tail -n 1 1>second.txt
+          echo "----------"
+          cat second.txt
+          if grep "balance: 0.000000000" second.txt; then
+            echo "Duplicated faucet rejected"
+          else
+            echo "Duplicated faucet not rejected!"
+            exit 1
+          fi
+        env:
+          SN_LOG: "all"
+        timeout-minutes: 5
+
+      - name: Create and fund a wallet with different keypair
+        run: |
+          ls -l /home/runner/.local/share
+          ls -l /home/runner/.local/share/safe
+          rm -rf /home/runner/.local/share/safe/test_faucet
+          rm -rf /home/runner/.local/share/safe/test_genesis
+          rm -rf /home/runner/.local/share/safe/client
+          GENESIS_PK=a9925296499299fdbf4412509d342a92e015f5b996e9acd1d2ab7f2326e3ad05934326efdc345345a95e973ac1bb6637 GENESIS_SK=40f6bbc870355c68138ac70b450b6425af02b49874df3f141b7018378ceaac66 ~/faucet --log-output-dest=data-dir send 100000000 $(~/safe --log-output-dest=data-dir wallet address | tail -n 1) | tail -n 1 1>different.txt
+          echo "----------"
+          cat different.txt
+          if grep "balance: 0.000000000" different.txt; then
+            echo "Faucet with different key rejected"
```

Comment:
is this `Faucet` key or GENESIS key. I'm a bit confused here

Reply:
It's Faucet using defferent genesis key
got it updated